app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java

```diff
@@ -0,0 +1,294 @@
+            this.mProgressDialog.setCancelable(false);
+            this.mProgressDialog.show();
+        }
+
+        @Override
+        protected void onPostExecute(Void unused) {
+            super.onPostExecute(unused);
+            // Check progress dialog
+            if (this.mProgressDialog != null) {
+                this.mProgressDialog.dismiss();
+            }
+        }
+
+        private void importHosts(AppDatabase database, Set<String> hosts) {
+            for (String url: hosts) {
+                HostsSource source = new HostsSource();
+                source.setUrl(url);
+                source.setEnabled(true);
+                AppExecutors.getInstance().diskIO().execute(() -> database.hostsSourceDao().insert(source));
+            }
+        }
+    }
+
+    /**
+     * This class is an {@link AsyncTask} to export user lists.
+     *
+     * @author Bruce BUJON (bruce.bujon(at)gmail(dot)com)
+     */
+    private static class ExportListsTask extends AsyncTask<Void, Void, Boolean> {
+        /**
+         * A weak reference to application context.
+         */
+        private final WeakReference<Context> mWeakContext;
+        /**
+         * The progress dialog.
+         */
+        private ProgressDialog mProgressDialog;
+
+        /**
+         * Constructor.
+         *
+         * @param context The application context.
+         */
+        private ExportListsTask(Context context) {
+            // Store context into weak reference to prevent memory leak
+            this.mWeakContext = new WeakReference<>(context);
+        }
+
+        @Override
+        protected Boolean doInBackground(Void... unused) {
+            // Get context from weak reference
+            Context context = this.mWeakContext.get();
+            if (context == null) {
+                // Fail to export
+                return false;
+            }
+            // Get database
+            AppDatabase database = AppDatabase.getInstance(context);
+            HostsSourceDao hostListItemDao = database.hostsSourceDao();
+            // Get list values
+            List<HostsSource> hosts = hostListItemDao.getEnabled();
+            // Check if sdcard can be written
+            File sdcard = Environment.getExternalStorageDirectory();
+            if (!sdcard.canWrite()) {
+                Log.e(Constants.TAG, "External storage can not be written.");
+                // Fail to export
+                return false;
+            }
+            // Create export file
+            File exportFile = new File(sdcard, "adaway-hosts-export");
+            // Open writer on the export file
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(exportFile))) {
+                writer.write(Constants.HEADER_EXPORT + Constants.LINE_SEPARATOR);
+                // Write blacklist items
+                for (HostsSource host : hosts) {
+                    writer.write( host.getUrl() + Constants.LINE_SEPARATOR);
```

Comment:
Better use `writer.newLine();` than custom `LINE_SEPARATOR` constant (I try to get rid of it).